src/Reflection/Adapter/ReflectionProperty.php

```diff
@@ -86,20 +75,17 @@ public function setValue($object, $value = null)
         }
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function hasType()
+    public function hasType(): bool
     {
         return $this->betterReflectionProperty->hasType();
     }
 
     /**
-     * {@inheritDoc}
+     * @psalm-suppress MismatchingDocblockReturnType
      */
-    public function getType()
+    public function getType(): ReflectionUnionType|ReflectionNamedType|ReflectionType|null
```

Comment:
`ReflectionType` should not be part of this union type

Reply:
This applies to any case where we see `ReflectionUnionType|ReflectionNamedType|ReflectionType`: the actual type should be as strict as possible, and `ReflectionType` is too wide.

Reply:
Ah, perhaps my misconception: `ReflectionType` still applies for `object`, `array` and other core types, right?

Reply:
I'm really not sure... I was sometimes just "trial and error" to make Psalm/PHPStan/PHPUnit/PHP happy...

Core types are `ReflectionNamedType`: https://3v4l.org/NOrU0